codeowners: Set Dev-Inf as codeowner of diagrams.go

nickvigilante opened the pull request:
Dev-Inf/Docs Engineering is now the owner of diagrams.go.

Release note: None

Release justification: non-production code change

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/87427)
<!-- Reviewable:end -->

rickystewart commented on .github/CODEOWNERS:
```diff
@@ -237,7 +237,7 @@
 /pkg/cmd/compile-build/      @cockroachdb/dev-inf
 /pkg/cmd/cr2pg/              @cockroachdb/sql-experience
 /pkg/cmd/dev/                @cockroachdb/dev-inf
-/pkg/cmd/docgen/             @cockroachdb/docs
```
Hmm, I think you want to keep this line so that the other files in the `docgen` directory still have the correct owners.

nickvigilante replied:
I'm not yet sure who owns that whole folder, but I know that I own `diagrams.go`, and docs is not the owner of the whole folder. I updated it to be unowned for the time being.

rickystewart replied:
CI might not let your PR through due to the unowned files. Let's wait and see if it passes.

Commit:
codeowners: Set Dev-Inf as codeowner of diagrams.go

Dev-Inf/Docs Engineering is now the owner of diagrams.go.

Release note: None

Release justification: non-production code change
